include/nbl/builtin/hlsl/matrix_utils/transformation_matrix_utils.hlsl

```diff
@@ -1,15 +1,45 @@
-#ifndef _NBL_BUILTIN_HLSL_TRANSFORMATION_MATRIX_UTILS_INCLUDED_
-#define _NBL_BUILTIN_HLSL_TRANSFORMATION_MATRIX_UTILS_INCLUDED_
+#ifndef _NBL_BUILTIN_HLSL_MATRIX_UTILS_TRANSFORMATION_MATRIX_UTILS_INCLUDED_
+#define _NBL_BUILTIN_HLSL_MATRIX_UTILS_TRANSFORMATION_MATRIX_UTILS_INCLUDED_
 
 #include <nbl/builtin/hlsl/cpp_compat.hlsl>
+#include <nbl/builtin/hlsl/math/quaternion/quaternion.hlsl>
+// TODO: remove this header when deleting vectorSIMDf.hlsl
+#include <nbl/core/math/glslFunctions.h>
+#include "vectorSIMD.h"
 
 namespace nbl
 {
 namespace hlsl
 {
 
+// TODO: if `IdentityFloat32_t3x4` and `IdentityFloat32_t3x4` constexprs are ok, then I can expand them into templated struct, not doing it untill the concept is approved
+//template<typename T, uint32_t N, uint32_t M>
+//struct IdentityMatrix
+//{
+//
+//};
+NBL_CONSTEXPR hlsl::float32_t3x4 IdentityFloat32_t3x4 =
+	hlsl::float32_t3x4(hlsl::float32_t4(1, 0, 0, 0), hlsl::float32_t4(0, 0, 1, 0), hlsl::float32_t4(0, 0, 1, 0));
+NBL_CONSTEXPR hlsl::float32_t4x4 IdentityFloat32_t4x4 =
+	hlsl::float32_t4x4(hlsl::float32_t4(1, 0, 0, 0), hlsl::float32_t4(0, 0, 1, 0), hlsl::float32_t4(0, 0, 1, 0), hlsl::float32_t4(0, 0, 0, 1));
+
+// TODO: this is temporary function, delete when removing vectorSIMD
 template<typename T>
-matrix<T, 4, 4> getMatrix3x4As4x4(const matrix<T, 3, 4>& mat)
+inline core::vectorSIMDf transformVector(const matrix<T, 4, 4>& mat, const core::vectorSIMDf& vec)
+{
+	core::vectorSIMDf output;
```

Comment:
https://github.com/Devsh-Graphics-Programming/Nabla/pull/760/files#r1828132005

Reply:
I will remove every occurance of the simd vector when I work on it, no need to spam xd